### DISCUSSION TITLE
Welcome to Gwion Discussions!

### DISCUSSION THREAD

**Discussion by fennecdjay** [Gwion/Gwion#224] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by scalarwaves** [comment]
Hi Jérémie,
Hello all,

I've been following the Gwion project for about 2 years, and I've been using ChucK for much longer. I am very interested in systems programming and the Rust language in particular. I proposed a bridge between ChucK and Gwion and I would like to explore more about this.

Look forward to hearing more ideas about this amazing language.

**Reply by fennecdjay** [reply]
Hi Andrew, I think about this bridge almost everyday TBH.

I have one thing left in Gwion before I finish `ck2gw` (I already know how to get chuck's AST, so all that's left is to write a pass that outputs gwion code): I want to introduce Optionnal types or something of that kind.
I'm also thinking of dropping gotos, but that's less important.

Right now there is a need for such a program, as there are some gwion users on the programming language discord, and we could really use such a tool.

We now have a near complete port of STK in Gwion! Given we already have Soundpipe and a few other libs, that's a whole lot of UGens :smile:

I you have more ideas for this bridge or gwion, I'll be glad to hear them.

Looking forward to read you, here or on [discord](https://discord.gg/4Kjt3ZE) (there is a #gwion channel).

**Comment by redshift-cyber** [comment]
Hello

**Reply by fennecdjay** [reply]
Hello :)

**Reply by redshift-cyber** [reply]
hi
